core/util.go

```diff
@@ -395,6 +399,15 @@ func IsASCII(str string) bool {
 	return true
 }
 
+// IsCanceled returns true if err is non-nil and is either context.Canceled, or
+// has a grpc code of Canceled. This is useful because cancelations propagate
+// through gRPC boundaries, and if we choose to treat in-process cancelations a
+// certain way, we usually want to treat cross-process cancelations the same
+// way.
+func IsCanceled(err error) bool {
```

Comment:
I see you've moved this into `core`. Why's that? In general, software tends to accumulate miscellaneous items in a `util` directory (or in our case `core`), which IMO is a bit of an anti-pattern. Packages are cheap, so it's fine to have small single purpose ones.

Reply:
Discussed offline. Samantha explained that when she's looking for a utility function that's used across multiple packages, she expects to find it in core (like, for instance, [`IsAnyNilOrZero()`](https://github.com/letsencrypt/boulder/pull/4971)). I'm okay doing this move on that basis, so closing this comment.